.github/workflows/precommit-windows.yml

```
@@ -0,0 +1,198 @@
+        if: always()
+        uses: ./.github/workflows/pr-sccache-save
+
+      - name: Package Build Directory
+        shell: bash
+        run: |
+          tar -c . | zstd -T0 -c > ../llvm-project.tar.zst
+          mv ../llvm-project.tar.zst .
+
+      - name: Upload Build
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 #v4.3.0
+        with:
+          name: part1
+          path: llvm-project.tar.zst
+          retention-days: 2
+
+  build-windows-part2:
+    name: "Build All"
+    needs:
+      - build-windows-part1
+    runs-on: windows-2022
+    steps:
+      - name: Setup Windows
+        uses: llvm/actions/setup-windows@main
+        with:
+          arch: amd64
+
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          max-size: 2G
+          variant: sccache
+          key: precommit-windows
+          # The llvm-project.zip archive contains the sccache directory from
+          # part 1, so we don't need to restore the cache.
+          restore: false
+
+      - name: Download Artifact
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+        with:
+          pattern: part1
+
+      - name: Unpack Artifact
+        shell: bash
+        run: |
+          tar --zstd -xf llvm-project.tar.zst
+          rm llvm-project.tar.zst
+
+      - name: Build
+        shell: bash
+        run: |
+          ls
+          ninja -C build
+
+      - name: Save sccache for next PR run
+        if: always()
+        uses: ./.github/workflows/pr-sccache-save
+ 
+      - name: Package Build Directory
+        shell: bash
+        run: |
+          tar -c . | zstd -T0 -c > ../llvm-project.tar.zst
+          mv ../llvm-project.tar.zst .
+
+      - name: Upload Build
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 #v4.3.0
+        with:
+          name: part2
+          path: llvm-project.tar.zst
+          retention-days: 2
+
+  test-windows:
+    name: "Test"
+    needs:
+      - compute-projects
+      - build-windows-part2
+    runs-on: windows-2022
+    steps:
+      - name: Setup Windows
+        uses: llvm/actions/setup-windows@main
+        with:
+          arch: amd64
+
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          max-size: 2G
+          variant: sccache
+          key: precommit-windows
+          # The llvm-project.zip archive contains the sccache directory from
+          # part 1, so we don't need to restore the cache.
+          restore: false
+
+      - name: Download Artifact
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+        with:
+          pattern: part2
+          merge-multiple: true
+
+      - name: Unpack Artifact
+        shell: bash
+        run: |
+          tar --zstd -xf llvm-project.tar.zst
+          rm llvm-project.tar.zst
+
+      - name: Check LLVM
```

```suggestion
      - name: Check LLVM
```

This checks more than LLVM, doesn't it?